src/api/spec/models/bs_request_spec.rb

```diff
@@ -68,4 +68,30 @@
       end
     end
   end
+
+  describe '#changestate' do
+    let(:request) { create(:bs_request) }
+    let(:admin) { create(:admin_user) }
+
+    context 'for wiping request' do
+      before do
+        request
+
+        # assign the admin after the first call of 'request'. this is important to avoid errors
+        User.current = admin
```

Comment:
Btw: why does the user need to be an admin user? Have you tried ``User.current = user``?